### ISSUE TITLE
Change Request: Consider updating the Android minimum supported level guidelines

### ISSUE THREAD

**Issue by vcolin7** [Azure/azure-sdk#5674]
Thank you for contributing to the design guidelines for Azure SDKs.  We welcome all discussions and contributions.  Since the design guidelines affect the client libraries for all Azure services, we thoroughly discuss and vet any design guideline changes before we incorporate them into the official guidance.  This process can take some time, so your patience is appreciated.

** Before submitting, ensure you adjust the title of the issue appropriately **

## The Basics

* Which design guideline is affected?
  - General Guidelines ([link](https://azure.github.io/azure-sdk/android_design.html#general-guidelines))
  - Azure SDK Library Design
    - Supported Platforms ([link](https://azure.github.io/azure-sdk/android_design.html#android-library-support))

* Which target languages are affected?
Android

* Describe the change:
Update the verbiage of the minimum Android level guidelines to clarify that, while it's desirable to support an API level as low as 15, library owners can actually decide what level will be the baseline for their library.

## Reasoning
This has been a point of confusion in the past with library owners and without clarification, it may lead people to think they cannot have a higher baseline.

**Comment by JonathanGiles** [comment]
I agree with this proposal. @vcolin7 did you want to propose new text, or did you want to leave it to me?

**Comment by vcolin7** [comment]
@JonathanGiles I was thinking of leaving it up to you if you don't mind :)
